deltachat-ios/ViewModel/ContactDetailViewModel.swift

```diff
@@ -87,6 +88,10 @@ class ContactDetailViewModel {
             chatOptions.append(.locations)
         }
 
+        if #available(iOS 15, *) {
+            chatActions = [.addToHomescreen]
+        }
```

Comment:
this needs to go into the `if chatId != 0` branch below - as without a chatId, we cannot add/remove things from the widget

i would also move that _below_ "Archive", which is very probably by far the most often used action, and should be first

btw, this brings me to the question what happens when a chat is deleted - is the widget updated at least after some time? otherwise, we should file and issue for that

Reply:
I didn't check or looked at the code how it should behave, so please feel free to tackle #2457 :) Should be fairly straight forward

Reply:
> `if chatId != 0` 

and 

> as without a chatId

means that `0` counts as "no chatId"?
I'll move the code. And search for the group-related stuff.